src/org/openstreetmap/josm/gui/tagging/presets/TaggingPresetSelector.java

```diff
@@ -212,32 +218,30 @@ public TaggingPresetSelector(boolean displayOnlyApplicable, boolean displaySearc
         JPanel pnChecks = new JPanel();
         pnChecks.setLayout(new BoxLayout(pnChecks, BoxLayout.Y_AXIS));
 
-        if (displayOnlyApplicable) {
-            ckOnlyApplicable = new JCheckBox();
-            ckOnlyApplicable.setText(tr("Show only applicable to selection"));
-            pnChecks.add(ckOnlyApplicable);
-            ckOnlyApplicable.addItemListener(e -> filterItems());
-        } else {
-            ckOnlyApplicable = null;
-        }
-
-        if (displaySearchInTags) {
-            ckSearchInTags = new JCheckBox();
-            ckSearchInTags.setText(tr("Search in tags"));
-            ckSearchInTags.setSelected(SEARCH_IN_TAGS.get());
-            ckSearchInTags.addItemListener(e -> filterItems());
-            pnChecks.add(ckSearchInTags);
-        } else {
-            ckSearchInTags = null;
-        }
-
-        if (displayDeprecated) {
-            ckDeprecated = new JCheckBox();
-            ckDeprecated.setText(tr("Show deprecated tags"));
-            pnChecks.add(ckDeprecated);
-            ckDeprecated.addItemListener(e -> filterItems());
-        } else {
-            ckDeprecated = null;
+        for (PresetSearchFilter option : options) {
+            switch(option) {
+                case ONLY_APPLICABLE: {
+                    ckOnlyApplicable = new JCheckBox();
+                    ckOnlyApplicable.setText(tr(PresetSearchFilter.ONLY_APPLICABLE.getText()));
```

Comment:
You could use `option.getText()` here instead of calling the qualified name.

We want to have a map of option -> checkboxes, so `EnumMap<PresetSearchFilter, JCheckBox>` would work. Example:

```java
final Map<PresetSearchFilter, JCheckBox> checkboxes = new EnumMap<PresetSearchFilter, JCheckBox>(PresetSearchFilter.class);
for (PresetSearchFilterOption option : options) {
    final JCheckBox box = new JCheckBox();
    box.setText(tr(option));
    box.addItemListener(e -> filterItems());
    pnChecks.add(box);
    switch (option) {
    case SEARCH_IN_TAGS:
        box.setSelected(PREF_SEARCH_IN_TAGS.get()); 
        break;
        // Do other enum values
    }
}
```